Добавлен компонент регистрации и авторизации

JstUsername opened the pull request:
ඞ

JstUsername committed:
feat(breakpoints): брейкпоинты добавлены в тему

JstUsername committed:
feat(assets/login): добавлено изображение и иконки для компонента Login

JstUsername committed:
feat(Login.tsx): добавлен компонент регистрации и авторизации

JstUsername committed:
feat(Login): компонент регистрации и авторизации добавлен в App.tsx

JstUsername committed:
fix(StyledOutlinedButton): удалены лишние значения padding

VeronikaBulaeva commented on src/components/Login.tsx:
```diff
@@ -0,0 +1,279 @@
+}
+
+export default function Login({ typeLogin, setTypeLogin, setOpenLogin }: loginProps) {
+  const [showPassword, setShowPassword] = useState<boolean>(false);
```
Можно не прокидывать boolean в дженерик, т.к это примитивный тип

VeronikaBulaeva commented on src/components/Login.tsx:
```diff
@@ -0,0 +1,279 @@
+            </FormControl>
+            <AdditionalFields>
+              <Box sx={{ display: 'flex', gap: 1 }}>
+                <StyledCheckbox tabIndex={0} onClick={() => setChecked((prev) => !prev)}>
```
функции лучше выносить

majishanred replied:
она маленькая, можно не выносить

JstUsername committed:
feat(generics): удалены примитивные типы для стейтов

JstUsername committed:
fix(xs): значение для xs брейкпоинта изменено на 0

JstUsername committed:
fix(props): удалены лишние props при обращении к теме

u1timaD commented on src/components/Login.tsx:
```diff
@@ -0,0 +1,279 @@
+  const isSmallScreen = useMediaQuery(theme.breakpoints.down('sm'));
+
+  return (
+    <Grid
```
думаю, что то тут надо вынести на отдельные комоненты, большое полотно получается, не читабельно

majishanred replied:
Да, думаю надо разбить на 2 отдельных компонента, Login и Registration, а то так все в одну кучу смешалось.

JstUsername replied:
Будет очень много повторяющегося кода, в таком случае. Надо подумать...

u1timaD replied:
думаю мысль должны быть такая: Если компонент можо переиспользоваться, то выносим, если нет, то оставляем

u1timaD commented on src/components/Login.tsx:
```diff
@@ -0,0 +1,279 @@
+            )}
+            <FormControl>
+              <Typography variant="body1" sx={{ fontSize: '14px', marginBottom: '8px' }}>
+                Email
```
я сделал бы label с for чтобы при нажатии на строку email кидало на input

u1timaD replied:
и кажется странно по симантке, тянет максиму на span , но не на р

u1timaD commented on src/components/Login.tsx:
```diff
@@ -0,0 +1,279 @@
+              </Typography>
+            </Link>
+          ) : (
+            <Link href="#" underline="hover" color="primary.dark" onClick={() => setTypeLogin('registration')}>
```
![image](https://github.com/JstUsername/team-landing/assets/45046356/5639a1f6-6fe9-474a-85b7-c3ca409094cc)
может докапываюсь, но ссылка довольно длинная сursorPointer

u1timaD commented on src/components/Login.tsx:
```diff
@@ -0,0 +1,279 @@
+  text-transform: none;
+  font-size: 16px;
+  line-height: 1;
+  letter-spacing: 0.03em;
```
емы = зло )) я бы старался уходить от них)

JstUsername replied:
Можно, но если сделать в px, то получится 0.5. WebStorm ругается на дробное значение...

u1timaD replied:
тут надо спрашивать ваню думаю) я слшал что никто не слюбит em, а чем мы хуже)

majishanred commented on src/components/Login.tsx:
```diff
@@ -0,0 +1,279 @@
+                      src="/src/assets/login/icons/union.svg"
+                      alt="Union"
+                      style={{ position: 'absolute', left: '3px', top: '5px' }}
+                    />
```
Чекбоксы можно стайлить через css, тогда можно будет обойтись вообще без checked (странноватое замечание с учетом того что у меня это в коде тоже есть, но там другой случай)

JstUsername replied:
Тут дело в том что это не совсем чекбокс, а по-сути div c иконкой внутри.
Сделал так потому что проблематично переписывать что mui чекбокс, что стоковый HTML, чтобы он по-итогу выглядел как в макете.

majishanred replied:
Оки, тогда сделай так, чтобы можно было активировать чекбокс через клаву, а то он через таб выделяется, но никак не тригерится при нажатии пробела или enter.

majishanred commented on src/components/Login.tsx:
```diff
@@ -0,0 +1,279 @@
+        <img
+          src="/src/assets/login/images/login.webp"
+          alt="Decorative image from the login page"
+          style={{ width: '100%', height: '100%', objectFit: 'cover' }}
```
Денис сказал инлайн стили = плохо

JstUsername replied:
Придерживаюсь правила не больше 3 пропсов/параметров в sx/style.
Если на каждую такую стилизацию создавать переменные, то получится штук 20.

majishanred commented on src/components/Login.tsx:
```diff
@@ -0,0 +1,279 @@
+  const [showPassword, setShowPassword] = useState(false);
+  const [checked, setChecked] = useState(false);
+
+  const handleClickShowPassword = () => setShowPassword((show) => !show);
```
у тебя везде евенты обрабатываются через () => setЧтоНибудь, а тут она висит одинокая) Надо бы убрать тогда. Я у себя такой же прикол заметил, надо поправить.

JstUsername committed:
fix(handleClickShowPassword): setShowPassword вынесен в onClick

JstUsername committed:
feat(handleKeyDown): добавлена функция для переключения чекбокса по нажатию на клавиши

JstUsername committed:
feat(StyledLink): добавлены стили для Link исправляющие ширину

JstUsername committed:
fix(InputLabel): заголовки для input реализованы с помощью label

JstUsername committed:
fix(letter-spacing): удалены свойства letter-spacing

IvanKoigerov commented on src/App.tsx:
```diff
@@ -1,13 +1,22 @@
+          <Login typeLogin={typeLogin} setTypeLogin={setTypeLogin} setOpenLogin={setOpenLogin} />
+        ) : (
+          'Add the components here'
+        )}
```
тебе нужно делать отдельную страницу для этого)
не в app описывать

IvanKoigerov commented on src/components/Login.tsx:
```diff
@@ -0,0 +1,301 @@
+import useMediaQuery from '@mui/material/useMediaQuery';
+import theme from '../theme.tsx';
+
+interface loginProps {
+  typeLogin: string;
+  setTypeLogin: Dispatch<SetStateAction<string>>;
+  setOpenLogin: Dispatch<SetStateAction<boolean>>;
+}
```
интерфейсы пишуться с большой буквы

IvanKoigerov commented on src/components/Login.tsx:
```diff
@@ -0,0 +1,301 @@
+    <Grid
+      container
+      spacing={0}
+      sx={{
+        height: '100vh',
+        overflow: 'hidden',
+        [theme.breakpoints.down('sm')]: { height: '100%', overflow: 'auto' },
+      }}
+    >
```
выносим в styled такие вещи

IvanKoigerov commented on src/components/Login.tsx:
```diff
@@ -0,0 +1,301 @@
+    >
+      <Grid xs={12} sm={6}>
+        <LeftColumnWrapper>
+          <Typography variant={isSmallScreen ? 'h3' : 'h2'} sx={{ margin: 0 }}>
+            {typeLogin === 'registration' ? 'Регистрация' : 'Вход'}
+          </Typography>
```
в тему уносим

IvanKoigerov commented on src/components/Login.tsx:
```diff
@@ -0,0 +1,301 @@
+          <FormWrapper component="form">
+            {typeLogin === 'registration' && (
+              <Box sx={{ display: 'flex', gap: 2 }}>
+                <FormControl sx={{ flex: 1 }}>
+                  <StyledInputLabel shrink htmlFor="name-input">
+                    Имя
+                  </StyledInputLabel>
+                  <StyledTextField id="name-input" placeholder="Иван" />
+                </FormControl>
```
у тебя куча повторений этой конструкции, выноси в отдельный компонент

IvanKoigerov commented on src/components/Login.tsx:
```diff
@@ -0,0 +1,301 @@
+              <StyledInputLabel shrink htmlFor="email-input">
+                Email
+              </StyledInputLabel>
+              <StyledTextField id="email-input" placeholder="example@email.com" />
```
хотел бы чтобы ты сделал валидацию для инпута

IvanKoigerov commented on src/components/Login.tsx:
```diff
@@ -0,0 +1,301 @@
+  flex-direction: column;
+  justify-content: center;
+  gap: 48px;
+  @media (max-width: ${theme.breakpoints.values.sm}px) {
+    height: 100%;
+    gap: 32px;
+  }
+  @media (max-width: ${theme.breakpoints.values.md}px) {
+    padding: 32px 16px;
+  }
+  @media (min-width: ${theme.breakpoints.values.md + 1}px) {
+    padding: 80px;
```
не корректно тему юзаешь

IvanKoigerov commented on src/components/Login.tsx:
```diff
@@ -0,0 +1,301 @@
+  gap: 16px;
+`;
+
+const StyledTextField = styled(TextField)`
+  & .MuiInputBase-root {
+    border-radius: 0;
+    padding-right: 0;
+    font-size: 16px;
+    color: ${theme.palette.text.secondary};
+    & .MuiInputBase-input {
+      padding: 13px 16px;
+      background-color: ${theme.palette.secondary.light};
+      &::placeholder {
+        opacity: 1;
+      }
+    }
+    & fieldset {
+      border: none;
+      border-bottom: 1px solid #c1c7cd;
+    }
+    &:hover fieldset {
+      border-color: ${theme.palette.secondary.dark};
+    }
+    & .Mui-focused fieldset {
+      border-color: ${theme.palette.primary.main};
+      border-width: 1px;
+    }
+    & .MuiInputAdornment-root {
+      position: absolute;
+      right: 0;
+      padding-right: 16px;
+      margin: 0;
+    }
+  }
```
создавай отдельный компонент для инпутов и там вот это все закинь

IvanKoigerov commented on src/components/Login.tsx:
```diff
@@ -0,0 +1,301 @@
+              <StyledTextField id="email-input" placeholder="example@email.com" />
+            </FormControl>
+            <FormControl>
+              <StyledInputLabel shrink htmlFor="password-input">
+                Пароль
+              </StyledInputLabel>
```
и валидации пароля нет

JstUsername committed:
fix(theme): добавлен default вариант для background

JstUsername committed:
feat(svgr): добавлен и настроен плагин svgr

JstUsername committed:
fix(LoginProps): исправлено название интерфейса

JstUsername committed:
fix(theme): исправлено использование theme в styled-components

JstUsername committed:
fix(Login.tsx): компонент Login вынесен в отдельную папку

JstUsername committed:
fix(Login, App): исправлены импорты после переноса Login

JstUsername committed:
feat(Input.tsx): поля формы и их стили вынесены в отдельный компонент

JstUsername committed:
fix(Grid): стилизация Grid вынесена в styled-components

JstUsername committed:
feat(dependencies): в проект добавлен React Router

JstUsername committed:
feat(BrowserRouter): компонент App обернут в BrowserRouter

JstUsername committed:
feat(Routes, AppContent): добавлены пути, контент приложения вынесен в AppContent

JstUsername committed:
fix(routing): роутинг реализован с помощью React Router

IvanKoigerov commented on src/components/Login/Login.tsx:
```diff
@@ -0,0 +1,256 @@
+            <FormControl>
+              <Input inputId="email-input" inputLabel="Email" inputPlaceholder="example@email.com" />
+            </FormControl>
+            <FormControl>
+              <Input
+                inputId="password-input"
+                inputLabel="Пароль"
+                inputPlaceholder="******"
+                type={showPassword ? 'text' : 'password'}
+                InputProps={{
+                  endAdornment: (
+                    <InputAdornment position="end">
+                      <IconButton
+                        aria-label="toggle password visibility"
+                        onClick={() => setShowPassword((show) => !show)}
+                        edge="end"
+                      >
+                        <img
+                          src="/src/assets/login/icons/eye.svg"
+                          alt="Change password visibility"
+                          style={{ width: 20, height: 20 }}
+                        />
+                      </IconButton>
+                    </InputAdornment>
+                  ),
+                }}
+              />
+              {typeLogin === 'registration' && (
+                <PasswordSubtitle>
+                  Пароль должен содержать минимум 8 символов, строчные и прописные символы
+                </PasswordSubtitle>
+              )}
+            </FormControl>
```
Постарайся разбить по компоненам ещё чуть-чуть, чтобы проще читать код можно было 
У тебя есть вот пароль и ещё одна вещь которую было бы хорошо вынести))

IvanKoigerov commented on src/components/Login/Login.tsx:
```diff
@@ -0,0 +1,256 @@
+          <Box sx={{ display: 'flex', gap: 2, [theme.breakpoints.down('lg')]: { flexDirection: 'column' } }}>
+            <StyledOutlinedButton
+              variant="outlined"
+              startIcon={<img src="/src/assets/login/icons/google.svg" alt="Google icon" />}
+              onClick={goToHome}
+            >
+              <Typography sx={{ paddingX: 2, fontSize: '16px' }}>Вход с помощью Google</Typography>
+            </StyledOutlinedButton>
+            <StyledOutlinedButton
+              variant="outlined"
+              startIcon={<img src="/src/assets/login/icons/apple.svg" alt="Apple icon" />}
```
SVGR жду что тут использовать будешь 
А ещё в src никогда не пишут ссылку, импортируют svg или webp и вставляют имя

IvanKoigerov commented on src/components/Login/Login.tsx:
```diff
@@ -0,0 +1,256 @@
+      </Grid>
+      <Grid xs={0} sm={6}>
+        <img
+          src="/src/assets/login/images/login.webp"
```
Тут тоже

IvanKoigerov commented on src/components/Login/Login.tsx:
```diff
@@ -0,0 +1,256 @@
+          </Box>
+          <Divider />
+          {typeLogin === 'registration' ? (
+            <StyledLink to="/login">
+              <Typography variant="body1" sx={{ fontSize: '14px' }}>
+                Нет аккаунта? Зарегистрироваться
+              </Typography>
+            </StyledLink>
```
Вот это тоже выноситься в отдельный компонент

IvanKoigerov commented on src/components/Input/Input.tsx:
А почему решил тут FormControl не закидывать?

JstUsername replied:
Потому что для FormControl используются разные sx пропсы.

IvanKoigerov commented on src/components/Login/Login.tsx:
```diff
@@ -0,0 +1,256 @@
+  flex-direction: column;
+  justify-content: center;
+  gap: 48px;
+  @media (max-width: ${({ theme }) => theme.breakpoints.values.sm}px) {
+    height: 100%;
+    gap: 32px;
+  }
+  @media (max-width: ${({ theme }) => theme.breakpoints.values.md}px) {
+    padding: 32px 16px;
+  }
+  @media (min-width: ${({ theme }) => theme.breakpoints.values.md + 1}px) {
+    padding: 80px;
+  }
```
есть же theme.breakpoints.up() и theme.breakpoints.down()

JstUsername committed:
feat(commons): компонент Input переименован в FormInput и вынесен в commons

JstUsername committed:
fix(Input): исправлен иморт, Input => FormInput

JstUsername committed:
fix(Login): компонент переименован в LoginPage

JstUsername committed:
fix(FormInput): исправлено название интерфейса и компонента

JstUsername committed:
feat(HomePage.tsx): добавлен компонент для размещения основного контента приложения

JstUsername committed:
fix(import, routing): исправлены импорты и пути для роутинга, AppContent заменен на HomePage

JstUsername committed:
fix(BrowserRouter): BrowserRouter вынесен из main в App

JstUsername committed:
fix(icons): иконки отображены с помощью svgr

JstUsername committed:
fix(login.webp): logn.webp импортирован как модуль, исправлен src

JstUsername committed:
fix(@media): все медиа запросы переписаны на theme.breakpoints.(down/up)

JstUsername committed:
fix(responsive h2): изменение размера h2, в зависимости от брейкпоинта sm вынесено в theme

JstUsername committed:
fix(StyledLink): StyledLink вынесен в компонент FormLink

JstUsername committed:
fix(StyledCheckbox): StyledCheckbox вынесен в компонент FormCheckbox

JstUsername committed:
feat(assets/login): добавлена иконка закрытого глаза

JstUsername committed:
feat(FormPasswordInput): инпут для пароля вынесен из LoginPage в отдельный компонент.

Для FormPasswordInput добавлена иконка закрытого глаза при видимом пароле.

JstUsername committed:
fix(EyeCloseIcon): увеличен размер EyeCloseIcon и добавлен отступ

JstUsername committed:
feat(validation): добавлена валидация email и пароля.

Email валидируется на входе и регистрации, пароль валидируеся только на регистрации.

majishanred commented on src/theme.tsx:
```diff
@@ -86,4 +96,13 @@ export const theme = createTheme({
   },
 });
 
+theme.typography.h2 = {
+  [theme.breakpoints.up('sm')]: {
+    fontSize: theme.typography.h2.fontSize,
+  },
+  [theme.breakpoints.down('sm')]: {
+    fontSize: theme.typography.h3.fontSize,
+  },
+};
+
```
Можно и не так категорично, выбери один брейкпоинт, на него повесь что-то, а перед этим просто разбери объект theme.typography.h2. Ох чую нас ждет боль с этими хедерами)

JstUsername committed:
fix(theme): переопределена типизация темы, добавлен variant="formHeader"

IvanKoigerov commented on src/commons/FormInput/FormInput.tsx:
Мы же обсуждали, что нужно разделить, обычный инпут и password
Ну и ещё и email тогда тоже выноси
У тебя должен быть общий инпут со стилями, без особой логики
И вот в данной ситуации ты можешь ещё вынести `PasswordInput` и `EmailInput` - это отдельные компоненты

Да ты вынес пароль в отдельный компонент, но тут есть такие вещи как `password-input`, такого в общем компоненте не должно быть

IvanKoigerov commented on src/commons/FormInput/FormInput.tsx:
```diff
@@ -0,0 +1,103 @@
+    font-size: 16px;
+    color: ${({ theme }) => theme.palette.text.secondary};
+    & .MuiInputBase-input {
+      padding: 13px 16px;
```
давайте четные циферки использовать

IvanKoigerov commented on src/commons/FormPasswordInput/FormPasswordInput.tsx:
```diff
@@ -0,0 +1,66 @@
+  const [error, setError] = useState(false);
+
+  const validatePassword = (password: string) => {
+    const re = /^(?=.*[a-z])(?=.*[A-Z0-9]).{8,}$/;
```
не очень круто так сильно сокращать название переменных
И обычно регекспы выносят в utils, потому что много где они могут пригодиться

IvanKoigerov commented on src/components/Login/LoginPage.tsx:
```diff
@@ -0,0 +1,168 @@
+`;
+
+const StyledButton = styled(Button)`
+  padding: 14px 12px 14px 12px;
```
разве не можешь `padding: 14px 12px`

IvanKoigerov commented on src/components/Login/LoginPage.tsx:
```diff
@@ -0,0 +1,168 @@
+          <FormWrapper component="form">
+            {typeLogin === 'registration' && (
+              <Box sx={{ display: 'flex', gap: 2 }}>
+                <FormControl sx={{ flex: 1 }}>
```
Как уже говорил, выноси FormControl в твой дефолтный инпут

IvanKoigerov reviewed (commented):
Насчет по растановки папок)
Смотрите Page обычно не кладут в компонентс, а выносят в папку pages 
А вот остальные компоненты как раз закидывают в папку Components
Ну и у тебя все компоненты связаны с формой, соответсвенно закидываем так 
`/componets/form`

JstUsername committed:
fix(padding): исправлены отступы в FormInput и LoginPage

JstUsername committed:
fix(FormControl): FormControl вынесен в FormInput, для FormInput добавлен inputSx

JstUsername committed:
feat(utils/validate): функции для валидации email и пароль вынесены в utils/validate

JstUsername committed:
feat(FormEmailInput): добавлен отдельный компонент для email.

FormInput очищен от стейтов и условий, в FormPassword добавлен PasswordSubtitle.

JstUsername committed:
fix(file structure): страницы перемещены в pages, содержимое commons перемещено в components/form

JstUsername committed:
fix(import): исправлены импорты после перемещения commons, LoginPage, HomePage

IvanKoigerov commented on src/utils/validate.ts:
```diff
@@ -0,0 +1,9 @@
+export const validateEmail = (email: string) => {
+  const regexpEmail = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  return regexpEmail.test(email.toLowerCase());
+};
+
+export const validatePassword = (password: string) => {
+  const regexpPassword = /^(?=.*[a-z])(?=.*[A-Z0-9]).{8,}$/;
+  return regexpPassword.test(password);
+};
```
Обычно именно const regexpEmail = /^[^\s@]+@[^\s@]+\.[^\s@]+$/; выделяют, а не всю функцию, но это на будущие, ща сливаем
